server/api/operator.go

```diff
@@ -337,18 +338,33 @@ func (h *operatorHandler) Delete(w http.ResponseWriter, r *http.Request) {
 	h.r.JSON(w, http.StatusOK, "The pending operator is canceled.")
 }
 
-func parseStoreIDs(v interface{}) (map[uint64]struct{}, bool) {
-	items, ok := v.([]interface{})
+func parseStoreIDsAndPeerRoles(rawStoreIDs interface{}, rawPeerRoles interface{}) (map[uint64]placement.PeerRoleType, bool) {
+	storeIDs, ok := rawStoreIDs.([]interface{})
 	if !ok {
 		return nil, false
 	}
-	ids := make(map[uint64]struct{})
-	for _, item := range items {
-		id, ok := item.(float64)
+
+	var peerRoles []interface{}
+	peerRoles, ok = rawPeerRoles.([]interface{})
+	if !ok {
+		peerRoles = nil
+	}
+
+	hasRoles := len(peerRoles) >= len(storeIDs)
```

Comment:
I think it would be better to accept only zero length or the same length as storeID.

Reply:
Done.